source/extensions/quic_listeners/quiche/envoy_quic_server_stream.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+#include "quiche/quic/core/http/quic_spdy_server_stream_base.h"
+
+#pragma GCC diagnostic pop
+
+#include "extensions/quic_listeners/quiche/envoy_quic_stream.h"
+
+namespace Envoy {
+namespace Quic {
+
+// This class is a quic stream and also a response encoder.
```

Comment:
Ok, I'm going to take a break here as my review is getting less and less thoughtful. More tomorrow!  :-)